### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by IraGorfiniak**

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (Igor-Hurskyi) [inline .idea/vcs.xml:6]: should be ignored with .gitignore

File: src/FirstTask.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class FirstTask {
+    public FirstTask() {
+    }
+    public static void main(String[] args) {
+        List<String> FirstList = new ArrayList();
```
Comment (Igor-Hurskyi) [inline src/FirstTask.java:8]: variable names should start with lowercase, plus we should use diamond syntax in the constructor when it comes to collections - "<>", e.g.:
`List<String> firstList = new ArrayList<>();`

File: src/FirstTask.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        FirstList.add("Element 3");
+        FirstList.add("Element 1");
+        FirstList.add("Element 4");
+        FirstList.add("Element 1");
+        FirstList.add("Element 5");
+        FirstList.add("Element 6");
+        FirstList.add("Element 7");
+        FirstList.add("Element 8");
+        FirstList.add("Element 9");
+        FirstList.add("Element 1");
```
Comment (Igor-Hurskyi) [inline src/FirstTask.java:20]: You can use addAll() method if you have a lot of elements to put, so it all can be done in one line, e.g.:
`numbers.addAll(Arrays.asList(1, 2, 3, 4, 5));`

File: src/FirstTask.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        System.out.println("Original List" + FirstList);
+        int size = FirstList.size();
+
+        for(int i = 0; i < size / 2; ++i) {
```
Comment (Igor-Hurskyi) [inline src/FirstTask.java:24]: try not to use ++i in the loops if there's no good reason. use i++ instead

File: src/FirstTask.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            if (((String)FirstList.get(i)).equals("Element 1")) {
```
Comment (Igor-Hurskyi) [inline src/FirstTask.java:25]: (String) casting is redundant

Comment (Igor-Hurskyi) [reply src/FirstTask.java:25]: what were you trying to cast there?

File: src/FirstTask.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class FirstTask {
+    public FirstTask() {
+    }
```
Comment (Igor-Hurskyi) [inline src/FirstTask.java:6]: why did you created a constructor there?

File: src/FirstTask.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        int size = FirstList.size();
+
+        for(int i = 0; i < size / 2; ++i) {
+            if (((String)FirstList.get(i)).equals("Element 1")) {
+                FirstList.set(i, "New Element");
+            }
+        }
+
+        System.out.println("Updated List" + FirstList);
+    }
```
Comment (Igor-Hurskyi) [inline src/FirstTask.java:31]: program is doing what it meant to do
but why did you used Strings like "Element 1"... ?
looks weird
in that case it's better to use Integers, for example

File: src/SecondTask.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import java.util.Arrays;
+import java.util.HashSet;
+public class SecondTask {
+    public static void main(String[] args) {
+        Integer[] initialArray = {1, 2, 3, 3, 4, 5, 6, 3, 7, 8, 3, 9};
+        System.out.println("Original Array:" + Arrays.toString(initialArray));
+        HashSet<Integer> set = new HashSet<>(Arrays.asList(initialArray));
+        Integer[] updatedArray = set.toArray(new Integer[0]);
```
Comment (Igor-Hurskyi) [inline src/SecondTask.java:8]: you have just created an array with zero length (new Integer[0])
`Integer[] updatedArray = (Integer[]) set.toArray();` think would be better

File: src/SecondTask.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import java.util.Arrays;
+import java.util.HashSet;
+public class SecondTask {
+    public static void main(String[] args) {
+        Integer[] initialArray = {1, 2, 3, 3, 4, 5, 6, 3, 7, 8, 3, 9};
+        System.out.println("Original Array:" + Arrays.toString(initialArray));
+        HashSet<Integer> set = new HashSet<>(Arrays.asList(initialArray));
```
Comment (Igor-Hurskyi) [inline src/SecondTask.java:7]: `Set<Integer> set = new HashSet<>(Arrays.asList(initialArray));`
don't forget to use polymorphism benefits
it make our code more flexible

File: src/SecondTask.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+        Integer[] updatedArray = set.toArray(new Integer[0]);
+        System.out.println("Updated Array:" + Arrays.toString(updatedArray));
+        System.out.println("Updated Array printed in reverse order:");
+        for (int i = updatedArray.length - 1; i >= 0; i--)
+            System.out.print(updatedArray[i] + "  ");
```
Comment (Igor-Hurskyi) [inline src/SecondTask.java:12]: that should be the first part of the program

File: src/SecondTask.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (Igor-Hurskyi) [inline src/SecondTask.java:13]: try to separate some logical parts of the program with empty lines (just 1 empty line per separator will be enough

File: src/SecondTask.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.HashSet;
+public class SecondTask {
+    public static void main(String[] args) {
+        Integer[] initialArray = {1, 2, 3, 3, 4, 5, 6, 3, 7, 8, 3, 9};
+        System.out.println("Original Array:" + Arrays.toString(initialArray));
+        HashSet<Integer> set = new HashSet<>(Arrays.asList(initialArray));
+        Integer[] updatedArray = set.toArray(new Integer[0]);
+        System.out.println("Updated Array:" + Arrays.toString(updatedArray));
+        System.out.println("Updated Array printed in reverse order:");
+        for (int i = updatedArray.length - 1; i >= 0; i--)
```
Comment (Igor-Hurskyi) [inline src/SecondTask.java:11]: that is complicated.
try to stick closer to the default 'for' loop scheme - for(int i = 0; i < something; i++)

Comment (Igor-Hurskyi) [reply src/SecondTask.java:11]:
```
int lastIndex = updatedArray.length - 1;
for (int i = 0; i < updatedArray.length; i++) {
    System.out.print(updatedArray[lastIndex - i] + "  ");
}
```

Comment (Igor-Hurskyi) [reply src/SecondTask.java:11]: does it looks better? it does to me

File: src/ThirdTask.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+public class ThirdTask {
+    public static void main(String[] args) {
+        int[] initialArray = new int[]{10, 1, 3, 1, 5, 6, 1, 1, 8, 9};
```
Comment (Igor-Hurskyi) [inline src/ThirdTask.java:6]: you can just type that: `int[] initialArray = {10, 1, 3, 1, 5, 6, 1, 1, 8, 9};`

File: src/ThirdTask.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+        System.out.println("Array before update:"+Arrays.toString(initialArray));
+        int sum = IntStream.of(initialArray).sum();
```
Comment (Igor-Hurskyi) [inline src/ThirdTask.java:8]: smart

File: src/ThirdTask.java [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+        int minIndex = 0;
+        int minElement = initialArray[minIndex];
+        int average = sum / initialArray.length;
+        for (int j = 0; j < initialArray.length; j++) {
+            {
+                sum = sum + initialArray[j];
+            }
```
Comment (Igor-Hurskyi) [inline src/ThirdTask.java:15]: don't need that brackets

File: src/ThirdTask.java [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+    public static void main(String[] args) {
+        int[] initialArray = new int[]{10, 1, 3, 1, 5, 6, 1, 1, 8, 9};
+        System.out.println("Array before update:"+Arrays.toString(initialArray));
+        int sum = IntStream.of(initialArray).sum();
+        int minIndex = 0;
+        int minElement = initialArray[minIndex];
+        int average = sum / initialArray.length;
+        for (int j = 0; j < initialArray.length; j++) {
+            {
+                sum = sum + initialArray[j];
```
Comment (Igor-Hurskyi) [inline src/ThirdTask.java:14]: why do you need that? after that point your sum will be 90

File: src/SecondTask.java (hunk 8 above)
Comment (IraGorfiniak) [reply src/SecondTask.java:8]: tha is a way to convert a Java HashSet<Integer> to a primitive int array I have googled.  And HashSet<Integer> was used to keep unique values only

Comment (Igor-Hurskyi) [reply src/SecondTask.java:8]: do you have a link to it?
> tha is a way to convert a Java HashSet to a primitive int array I have googled.

File: src/FirstTask.java (hunk 2 above)
Comment (IraGorfiniak) [reply src/FirstTask.java:8]: Fixed in Homework 2

File: src/FirstTask.java (hunk 3 above)
Comment (IraGorfiniak) [reply src/FirstTask.java:20]: Fixed in Homework 2

File: src/FirstTask.java (hunk 4 above)
Comment (IraGorfiniak) [reply src/FirstTask.java:24]: Fixed in Homework 2

File: src/FirstTask.java (hunk 5 above)
Comment (IraGorfiniak) [reply src/FirstTask.java:25]: Removed in Homework 2

File: src/FirstTask.java (hunk 6 above)
Comment (IraGorfiniak) [reply src/FirstTask.java:6]: typo, removed in Homework 2

File: src/FirstTask.java (hunk 7 above)
Comment (IraGorfiniak) [reply src/FirstTask.java:31]: Fixed in Homework 2

File: src/ThirdTask.java (hunk 13 above)
Comment (IraGorfiniak) [reply src/ThirdTask.java:6]: Fixed in Homework 2

File: src/ThirdTask.java (hunk 14 above)
Comment (IraGorfiniak) [reply src/ThirdTask.java:8]: thanks

File: src/ThirdTask.java (hunk 15 above)
Comment (IraGorfiniak) [reply src/ThirdTask.java:15]: Fixed in Homework 2

File: src/ThirdTask.java (hunk 16 above)
Comment (IraGorfiniak) [reply src/ThirdTask.java:14]: I am not sure what was the purpose of this part, removed it in Fixed in Homework 2

File: src/SecondTask.java (hunk 12 above)
Comment (IraGorfiniak) [reply src/SecondTask.java:11]: Changed in Homework 2

**Pull request merged**